### ISSUE TITLE
After adding a card to a dashboard, refreshing the page adds it a second time

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#4473]
![out](https://cloud.githubusercontent.com/assets/1455846/23482612/8ebe6604-fe84-11e6-880e-04c9bf7ed8b7.gif)

1.  Add a Card to a Dashboard
2.  Dashboard goes into edit move to position the Card
2.  Position Card and hit "Save"
3.  Cool, I want to refresh my dashboard to update my query results
4.  Card is added to the Dashboard a *second* time and Dashboard is back in edit mode

It looks like it's happening because `?add=<card-id>` is part of the Dashboard URL. It probably makes sense to remove that from the URL after someone leaves Dashboard edit mode (after step 2)

**Comment by camsaul** [comment]
Another bit of weirdness is that when I refresh the page scrolls down automatically to the beginning of the dashboard (not sure if that's intentional, but it's jarring and you have to scroll back up to see the (now editable) title or the Cancel/Delete/Save buttons

**Comment by suhaildawood** [comment]
I've also experience something similar. When I've added a new card card to a dashboard and saved, routing to another dashboard puts me into edit mode with the previously added card.

**Comment by mazameli** [comment]
Hmm, I've been unable to reproduce any of these described behaviors in master. Are you still seeing this, Cam?

**Comment by camsaul** [comment]
@mazameli did you refresh the page when the URL still has the`?add=` bit in it?

**Comment by mazameli** [comment]
Okay, I figured it out — this only happens if you add a card to a dashboard from the _question_ page. It doesn't happen if you use the `add card` button from the _dashboard_ page.

**Comment by suhaildawood** [comment]
I believe this also happens when creating a new question (immediately adding it to a dashboard)

**Comment by tlrobinson** [comment]
We just need to make sure we remove that param after adding the card. Also we should switch to using the hash/fragment part of the url (`#?add=1234`) since query params are used for dashboard parameters.

**Comment by salsakran** [comment]
closed via #5038
